src/main/jbake/content/documentation/bundles/configuration-installer-factory.md

```diff
@@ -70,20 +76,52 @@ The quoted string format is equal to the definition from HTTP 1.1 ([RFC2616](htt
 
 The 1 character type code is one of:
 
-* `T` : simple string
-* `I` : Integer
-* `L` : Long
-* `F` : Float
-* `D` : Double
-* `X` : Byte
-* `S` : Short
-* `C` : Character
-* `B` : Boolean
+* `T` : `String`
+* `I` : `Integer`
+* `L` : `Long`
+* `F` : `Float`
+* `D` : `Double`
+* `X` : `Byte`
+* `S` : `Short`
+* `C` : `Character`
+* `B` : `Boolean`
+
+or for primitives
+
+* `i` : `int`
+* `l` : `long`
+* `f` : `float`
+* `d` : `double`
+* `x` : `byte`
+* `s` : `short`
+* `c` : `char`
+* `b` : `boolean`
 
-For Float and Double types the methods [Float.intBitsToFloat](https://docs.oracle.com/javase/7/docs/api/java/lang/Float.html#intBitsToFloat%28int%29) and [Double.longBitsToDouble](https://docs.oracle.com/javase/7/docs/api/java/lang/Double.html#longBitsToDouble%28long%29) are being used to convert the numeric string into the correct type. These methods rely on the IEEE 754 floating-point formats described in [Single-precision floating-point format](https://en.wikipedia.org/wiki/Single-precision_floating-point_format) and [Double-precision floating-point format](https://en.wikipedia.org/wiki/Double-precision_floating-point_format). A more user-friendly format is not yet supported for these types ([SLING-7757](https://issues.apache.org/jira/browse/SLING-7757)).
+For Float and Double types the methods [Float.intBitsToFloat](https://docs.oracle.com/javase/7/docs/api/java/lang/Float.html#intBitsToFloat%28int%29) and [Double.longBitsToDouble](https://docs.oracle.com/javase/7/docs/api/java/lang/Double.html#longBitsToDouble%28long%29) are being used to convert the numeric string into the correct type. These methods rely on the IEEE-754 floating-point formats described in [Single-precision floating-point format](https://en.wikipedia.org/wiki/Single-precision_floating-point_format) and [Double-precision floating-point format](https://en.wikipedia.org/wiki/Double-precision_floating-point_format). A more user-friendly format is not yet supported for these types ([SLING-7757](https://issues.apache.org/jira/browse/SLING-7757)).
+
+Multiple values enclosed by `[` and `]` lead to an array while those enclosed by `(` and `)` lead to a Collection in the underlying `java.util.Dictionary` of the `org.osgi.service.cm.Configuration` and vice-versa.
+
+Although both objects and primites are supported, both lead to generating the wrapper objects (and not the primitives) in the underlying `java.util.Dictionary` of the `org.osgi.service.cm.Configuration` during deserialization (even for multivalue entries).
 
 A number of such .config files exist in the Sling codebase and can be used as examples.
 
+#### Limitations
+
+* No support for Collections containing different types
+* No support for nested multivalues (arrays or Collections)
+* No write-support for primitive arrays
```

Comment:
Again, that was the point - there is support for primitive arrays. If the array is primitive it will be written out as lower case type and upper case otherwise.

Reply:
Maybe "write-support" is misleading, I actually meant the deserialized Dictionary will always contain arrays or collections of Objects, never of primitives (again look at http://svn.apache.org/viewvc/felix/releases/org.apache.felix.configadmin-1.8.12/src/main/java/org/apache/felix/cm/file/ConfigurationHandler.java?view=markup#l444)

Reply:
I think you are looking at the wrong place. The first deserialization is using the wrapper for the values. But the read for the array is using the primitive type so you end-up with a primitive array: http://svn.apache.org/viewvc/felix/releases/org.apache.felix.configadmin-1.8.12/src/main/java/org/apache/felix/cm/file/ConfigurationHandler.java?view=markup#l386